### PR TITLE
P20-749: Add the "Canvas Integration Early Access" block to the incubator page

### PR DESCRIPTION
**Pull request by artem-vavilov**
## Summary
The block will be displayed by default, but it can be removed by `DCDO.set('incubator-canvas-block-enabled', false)`

## Screenshot
incubator-canvas-integration

## Links
- JIRA ticket: [P20-749](https://codedotorg.atlassian.net/browse/P20-749)

### CODE REVIEW

File: apps/src/templates/studioHomepages/Incubator.jsx [hunk 1]
```diff
@@ -34,6 +44,27 @@ class Incubator extends Component {
...
+              description={i18n.incubator_canvasIntegration_earlyAccess_desc()}
+              marginBottom="20px"
+              buttons={[
+                {
+                  url: 'https://forms.gle/x7EBBiC18yJysb5D7',
+                  text: i18n.incubator_canvasIntegration_earlyAccess_signIn_button(),
+                  target: '_blank',
+                },
+                {
+                  url: '#',
```
Comment (artem-vavilov) [inline apps/src/templates/studioHomepages/Incubator.jsx]: The demo link will be provided by @alivira this week

File: apps/i18n/common/en_us.json [hunk 2]
```diff
@@ -1347,6 +1347,10 @@
   "incorrectAnswerUnmutableBody": "The answer you've entered is not correct.",
   "incorrectSolution": "Incorrect solution",
   "incorrectSolutionBody": "The solution you've entered is not correct.  Please try again!",
+  "incubator_canvasIntegration_earlyAccess_title": "Canvas Integration Early Access",
```
Comment (carl-codeorg) [inline apps/i18n/common/en_us.json:1357]: Is this the recommended naming convention for i18n strings to mimic a hierarchical structure? Curious why we don't support nested objects in our front-end strings when we do for our backend strings.

File: apps/src/templates/studioHomepages/Incubator.jsx [hunk 3]
```diff
@@ -2,8 +2,18 @@ import React, {Component} from 'react';
 import HeaderBanner from '../HeaderBanner';
 import TwoColumnActionBlock from '@cdo/apps/templates/studioHomepages/TwoColumnActionBlock';
 import Button from '@cdo/apps/templates/Button';
+import DCDO from '@cdo/apps/dcdo';
+import i18n from '@cdo/locale';
```
Comment (mgc1194) [inline apps/src/templates/studioHomepages/Incubator.jsx:10]: wow I can't believe we were not localizing strings here before.

**Review by mgc1194 — APPROVED** [review]
Looks great!
Thanks for adding theses changes!

File: apps/src/templates/studioHomepages/Incubator.jsx (hunk 1 above)
Comment (artem-vavilov) [reply apps/src/templates/studioHomepages/Incubator.jsx]: Replaced with the "Giudes" button

File: apps/i18n/common/en_us.json [hunk 4]
```diff
@@ -1353,6 +1353,10 @@
...
+  "incubator_canvasIntegration_earlyAccess_desc": "Try out our one-click Single Sign-On (SSO) and Roster Syncing integrations for Canvas now!",
+  "incubator_canvasIntegration_earlyAccess_signIn_button": "Sign-in now",
```
Comment (carl-codeorg) [inline apps/i18n/common/en_us.json]: Should this actually be "Sign up now"? The link goes to the EA form

File: apps/i18n/common/en_us.json (hunk 2 above)
Comment (artem-vavilov) [reply apps/i18n/common/en_us.json:1357]: Based on a discussion in one of our daily meetings, this is indeed the recommended naming convention for the i18n hierarchical structure.
@mgc1194, please correct me if I'm mistaken.

**Review by carl-codeorg — APPROVED** [review]
LGTM

File: apps/i18n/common/en_us.json (hunk 2 above)
Comment (artem-vavilov) [reply apps/i18n/common/en_us.json:1357]: @carl-codeorg I'm not sure, but I believe it's related to the convertation of i18n strings into JavaScript functions.

File: apps/src/lib/util/AnalyticsConstants.js [hunk 5]
```diff
@@ -220,6 +220,10 @@ const EVENTS = {
   RECOMMENDED_SIMILAR_CURRICULUM_SHOWN: 'Recommended Similar Curriculum Shown',
   RECOMMENDED_SIMILAR_CURRICULUM_CLICKED:
     'Recommended Similar Curriculum Clicked',
+
+  // LTI Incubator
+  LTI_INCUBATOR_SIGNUP_CLICKED: 'LTI_Incubator_SignUp_Clicked',
+  LTI_INCUBATOR_GUIDES_CLICKED: 'LTI_Incubator_Guides_Clicked',
```
Comment (nicklathe) [inline apps/src/lib/util/AnalyticsConstants.js]: Currently with other LTI event names, we've gone with lower_camel_case. I recommend converting to that so we are consistent.

File: apps/src/templates/studioHomepages/Incubator.jsx [hunk 6]
```diff
@@ -2,8 +2,36 @@ import React, {Component} from 'react';
...
+  }
+
+  reportEvent(eventName, platform = PLATFORMS.STATSIG) {
+    analyticsReporter.sendEvent(
+      eventName,
+      {
+        userId: this.state.currentUser.userId,
+        userType: this.state.currentUser.userType,
+        userName: this.state.currentUser.userName,
+        userRoleInCourse: this.state.currentUser.userRoleInCourse,
```
Comment (nicklathe) [inline apps/src/templates/studioHomepages/Incubator.jsx:28]: The way Statsig is configured, it [builds a user](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/lib/util/StatsigReporter.js#L23) object as part of the initialization process and includes it in the event log, so you don't need to add it here in the metadata. Also, I'm not sure we want to log the username, as I believe this is PII.

As a result, I'd remove everything here except userRoleInCourse.

**Review by nicklathe — COMMENTED** [review]
Left a couple comments.

File: apps/src/templates/studioHomepages/Incubator.jsx (hunk 6 above)
Comment (artem-vavilov) [reply apps/src/templates/studioHomepages/Incubator.jsx:28]: Got it, thanks!

File: apps/src/lib/util/AnalyticsConstants.js (hunk 5 above)
Comment (nicklathe) [reply apps/src/lib/util/AnalyticsConstants.js]: I mistyped. I meant snake_case, not camel_case. So for example: `lti_incubator_signup_clicked`.

**Review by nicklathe — APPROVED** [review]
:shipit:

File: apps/i18n/common/en_us.json (hunk 2 above)
Comment (mgc1194) [reply apps/i18n/common/en_us.json:1357]: It is related to what Artem said. I was trying to find where in the codebase we do the conversion of json to JS functions, but I could not get further than the [`safeLoadLocale`](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/util/safeLoadLocale.js).

Comment (mgc1194) [reply apps/i18n/common/en_us.json:1357]: Alternatively, we could have a separate file for LTI strings as we do for labs specific strings. But I prefer this approach.

**Review by mgc1194 — APPROVED** [review]

**Pull request merged**